tests/test_parser.py

```diff
@@ -21,44 +21,79 @@
+                                       data=None, headers=None)
 
-    @patch('requests.get')
-    def test_perform_get_request_returns_expected_text_on_valid_request(
-            self,
-            mock_response):
+    def test_perform_get_request_returns_expected_text_on_valid_request(self):
         url = 'https://smarsy.ua/'
-        mock_response.return_value.status_code = 200
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=200)
+            )
+        )
         expected_text = 'This is login Page'
-        mock_response(url).text = expected_text
-        self.assertEqual(perform_get_request(url), expected_text)
+        session.get(url).text = expected_text
+        self.assertEqual(perform_get_request(session, url), expected_text)
 
-    @patch('requests.get')
-    def test_perform_get_requestresponse_with_status_code_404_raises_exception(
-            self,
-            mock_response):
+    def test_perform_get_request_resp_with_status_code_404_raises_exception(
+            self):
         url = 'https://smarsy.ua/'
-        mock_response.return_value.status_code = 404
-        self.assertRaises(requests.HTTPError, perform_get_request, url)
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=404)
+            )
+        )
+        self.assertRaises(requests.HTTPError, perform_get_request, session,
+                          url)
+
+    def test_perform_get_request_uses_provided_data_for_get_request(
+            self):
+        expected_data = 'data'
+        expected_url = 'url'
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=200,
+                                  data=expected_data,
+                                  text=expected_url)
+            )
+        )
+        perform_get_request(session, expected_url, data=expected_data)
+        session.get.assert_called_with(url=expected_url, data=expected_data,
+                                       headers=None)
+
+    def test_perform_get_request_uses_provided_headers_for_get_request(
+            self):
+        expected_headers = {"a": 1}
+        expected_url = 'url'
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=200,
+                                  data=None,
+                                  text=expected_url,
+                                  headers=expected_headers)
+            )
+        )
+        perform_get_request(session, expected_url, headers=expected_headers)
+        session.get.assert_called_with(url=expected_url, data=None,
+                                       headers=expected_headers)
 
     def test_login_page_has_expected_title(self):
         html = '<html><title>Smarsy - Смарсі - Україна</title></html>'
         actual = validate_title(html)
         self.assertTrue(actual)
 
+
+class TestsPostPage(unittest.TestCase):
```

Comment:
the name is wrong. we do not post pages.

Reply:
TestsPostPage renamed to TestsPostRequest